### DISCUSSION TITLE
Rails - Load assets from gem

### DISCUSSION THREAD

**Discussion by Faq** [webpack/webpack#16910] [Q&A] [answered]
Hello,

Is there a way to load scss assets from gem when using `webpack 5` and `sass-loader` in scss file using `@import 'gem_name'`?
```
#webpack.config.js
{
        test: /\.(?:sa|sc|c)ss$/i,
        use: [MiniCssExtractPlugin.loader, 'css-loader', 'sass-loader'],
},
```
now resulting in:
```
ERROR in ./app/assets/stylesheets/application.scss
Module build failed (from ./node_modules/mini-css-extract-plugin/dist/loader.js):
HookWebpackError: Module build failed (from ./node_modules/sass-loader/dist/cjs.js):
SassError: Can't find stylesheet to import.

@import 'gem_name';
         ^^^^^^^^^^^^^^^^^^^^^^^^

```

Needed gem doesn't have npm package.

**Comment by alexander-akait** [comment, marked as answer]
It means you don't have `gem_name` in your dependecies or files

**Reply by Faq** [reply]
What dependencies or files you mean?

**Reply by alexander-akait** [reply]
What is `gem_name`?

**Reply by Faq** [reply]
It is private gem.

As an example gem https://github.com/thoughtbot/bourbon/tree/main/core
Lets say, if this one would not have npm package, how to load that [bourbon.scss](https://github.com/thoughtbot/bourbon/blob/main/core/_bourbon.scss) file?

**Reply by alexander-akait** [reply]
You can look at the `resolver` (look at `alias` or `roots`) option and setup it to resolve `gem_name`, I don't know there it is stored, so you need to find a path and set it, by default we are use `node_modules` for dependecies (because it is javascript), but I am thinking you use ruby gems so your path is different

**Reply by Faq** [reply]
Thank you for the information.
Gem asset path is different on local dev and on server, therefore need some function what would be looking for particular gem.
Decided to make npm package, what looks like better solution and in future can easily applied to other projects.
